internal/infoschema.go

```diff
@@ -48,6 +53,150 @@ func ProcessInfoSchema(conv *Conv, db *sql.DB) error {
 	return nil
 }
 
+// ProcessSqlData performs data conversion for source database
+// 'db'. For each table, we extract data using a "SELECT *" query,
+// convert the data to Spanner data (based on the source and Spanner
+// schemas), and write it to Spanner.  If we can't get/process data
+// for a table, we skip that table and process the remaining tables.
+//
+// Note that the database/sql library has a somewhat complex model for
+// returning data from rows.Scan. Scalar values can be returned using
+// the native value used by the underlying driver (by passing
+// *interface{} to rows.Scan), or they can be converted to specific go
+// types. Array values are always returned as []byte, a string
+// encoding of the array values. This string encoding is
+// database/driver specific. For example, for PostgreSQL, array values
+// are returned in the form "{v1,v2,..,vn}", where each v1,v2,...,vn
+// is a PostgreSQL encoding of the respective array value.
+//
+// We choose to do all type conversions explicitly ourselves so that
+// we can generate more targeted error messages: hence we pass
+// *interface{} parameters to row.Scan.
+func ProcessSqlData(conv *Conv, db *sql.DB) {
+	// TODO: refactor to use the set of tables computed by
+	// ProcessInfoSchema instead of computing them again.
+	tables, err := getTables(db)
+	if err != nil {
+		conv.unexpected(fmt.Sprintf("Couldn't get list of table: %s", err))
+		return
+	}
+	for _, t := range tables {
+		// PostgreSQL schema and name can be arbitrary strings.
+		// Ideally we would pass schema/name as a query parameter,
+		// but PostgreSQL doesn't support this. So we quote it instead.
+		q := fmt.Sprintf(`SELECT * FROM "%s"."%s";`, t.schema, t.name)
+		rows, err := db.Query(q)
+		if err != nil {
+			conv.unexpected(fmt.Sprintf("Couldn't get data for table: %s", err))
+			continue
+		}
+		defer rows.Close()
+		srcTable := buildTableName(t.schema, t.name)
+		srcCols, err1 := rows.Columns()
+		spTable, err2 := GetSpannerTable(conv, srcTable)
+		spCols, err3 := GetSpannerCols(conv, srcTable, srcCols)
+		spSchema, ok1 := conv.spSchema[spTable]
```

Comment:
These should be safe because:
a) spTable is a string, so if GetSpannerTable fails, it returns empty string. We will look up an empty string in conv.spSchema which will return not-found i.e. ok1 will be false.
b) if rows.Columns fails, srcCols will be the empty slice; GetSpannerCols will iterate over this and return an empty spCols.

Reply:
Based on our vc, we think we're good here. Resolving the thread.